core/trino-main/src/main/java/io/trino/event/QueryMonitor.java

```diff
@@ -703,6 +705,27 @@ private static StageCpuDistribution computeCpuDistribution(StageInfo stageInfo)
                 snapshot.getTotal() / snapshot.getCount());
     }
 
+    private static List<Optional<io.trino.spi.metrics.Distribution<?>>> getStageOutputBufferUtilizationDistributions(QueryInfo queryInfo)
+    {
+        if (queryInfo.getOutputStage().isEmpty()) {
+            return ImmutableList.of();
+        }
+
+        ImmutableList.Builder<Optional<io.trino.spi.metrics.Distribution<?>>> builder = ImmutableList.builder();
+        populateStageOutputBufferUtilizationDistribution(queryInfo.getOutputStage().get(), builder);
+
+        return builder.build();
+    }
+
+    private static void populateStageOutputBufferUtilizationDistribution(StageInfo stageInfo, ImmutableList.Builder<Optional<io.trino.spi.metrics.Distribution<?>>> distributions)
+    {
+        Optional<io.trino.spi.metrics.Distribution<?>> distribution = stageInfo.getStageStats().getOutputBufferUtilization().map(dist -> dist);
```

Comment:
The use of `Optional#map` here looks odd as it is being used to achieve a cast and not apply any actual mapping function.
Maybe we can change `Optional<io.trino.spi.metrics.Distribution<?>>` to `Optional<? extends io.trino.spi.metrics.Distribution<?>>` which will allow us to drop the unnecessary call to `Optional#map`.